config/webpack/CustomVersionFilePlugin.ts

```diff
@@ -0,0 +1,26 @@
+import fs from 'fs';
+import path from 'path';
+import type {Compiler} from 'webpack';
+import {version as APP_VERSION} from '../../package.json';
+
+/**
+ * Simple webpack plugin that writes the app version (from package.json) and the webpack hash to './version.json'
+ */
+class CustomVersionFilePlugin {
+    apply(compiler: Compiler) {
+        compiler.hooks.done.tap(this.constructor.name, () => {
+            const versionPath = path.join(__dirname, '/../../dist/version.json');
+            fs.mkdir(path.dirname(versionPath), {recursive: true}, (dirErr) => {
+                if (dirErr) {
+                    console.error('Error creating version.json directory', dirErr);
+                    return;
```

Comment:
```suggestion
                if (dirErr) {
                    throw dirErr;
                }
```

throw instead of console.error because console.error will keep the exit code 0 (it should be non-zero so we know something went wrong)